app/controllers/carts_controller.rb

```diff
@@ -51,4 +59,23 @@ def delete_cart
 
     cart_current.delete(params[:product_detail_id])
   end
+
+  def update_quantity_cart product_detail_id, quantity, product_quantity
+    if cart_current.key?(product_detail_id)
+      cart_current[product_detail_id] = if product_quantity < quantity.to_i
+                                          Settings.number.digits_1
+                                        else
+                                          quantity.to_i
+                                        end
+    else
+      cart_current[product_detail_id] = Settings.number.digits_1
+    end
+  end
+
+  def check_quantity_exceeds_amount quantity, product_quantity
+    return unless product_quantity < quantity.to_i
+
+    flash[:danger] = t "carts.update..exceed"
+    render js: "window.location = '#{cart_path}'"
```

Comment:
Nên báo cho ng dùng là sản phẩm nào đang bị out of stock nữa em
-> tức là trong message kèm tên sản phẩm vô

Reply:
dạ em đã fix ạ